### ISSUE TITLE
why softmax before celoss?

### ISSUE THREAD

**Issue by howardchina** [ArdhenduBehera/cap#13]
https://github.com/ArdhenduBehera/cap/blob/cae53050af583c2b3fa38081c0abc227199e7dc1/train_CAP.py#L159

Hi, I feel strange about this 'trick'?

**Comment by ArdhenduBehera** [comment]
Thanks
That's the final layer to compute the class probability. It is a standard layer for multi class classification problem.

**Comment by howardchina** [comment]
> 

It still confuse me. I thought celoss already made softmax inside, so we don't need additional softmax layer.

**Comment by howardchina** [comment]
> 

thx! it solved. https://www.tensorflow.org/api_docs/python/tf/keras/losses/CategoricalCrossentropy

**Comment by howardchina** [comment]
> Thanks That's the final layer to compute the class probability. It is a standard layer for multi class classification problem.

softmax is auto calculated by celoss in Pytorch, but not in TF. 
so we need softmax the input before celoss in TF.
